pandas/core/series.py

```diff
@@ -2530,6 +2530,13 @@ def append(self, to_append, ignore_index=False, verify_integrity=False):
             to_concat.extend(to_append)
         else:
             to_concat = [self, to_append]
+        for x in to_concat[1:]:
```

Comment:
can you make this a single check with `any(...)`

ideally annotate the args above while we're here

Reply:
According to @simonjayhawkins [here](https://github.com/pandas-dev/pandas/pull/31036#discussion_r367839776), `for(...)` fails faster. I guess it makes sense because with `if any(isinstance(x, ABCDataFrame) for x in to_concat[1:]):` it waits till the list comprehension is built.

Reply:
With `to_append` annotated using `Union[Series, Sequence[Series]]`, the static analysis is throwing this error:
**pandas/core/series.py:2539: error: List item 1 has incompatible type "Union[Any, Sequence[Any]]"; expected "Series"
Found 1 error in 1 file (checked 880 source files)**
Is there a better type or way to annotate the args?

Reply:
> waits till the list comprehension is built

i think that would be true for `any([isinstance(x, ABCDataFrame) for x in to_concat[1:]])`, but shouldnt be for `any(isinstance(x, ABCDataFrame) for x in to_concat[1:])`.

@simonjayhawkins am i missing something?

Reply:
Yeah let's go with `any(isinstance(x, ABCDataFrame) for x in to_concat[1:])`. I just timed all 3 and this one was the fastest. 
What do you think we should print in the msg? `to_append.__name__`, which will give sequence type if a sequence with DataFrame is passed? or just `DataFrame`?